src/python/WMCore/WMSpec/Steps/Executors/StageOut.py

```diff
@@ -136,48 +135,40 @@ def execute(self, emulator = None):
-                            logging.error(str(ex))
-                            logging.debug(fileName)
-                            logging.debug("minMergeSize: %s" % self.step.output.minMergeSize)
-                            manager.cleanSuccessfulStageOuts()
-                            stepReport.addError(self.stepName, 60401, "DirectToMergeFailure", str(ex))
-                    elif getattr(self.step.output, 'maxMergeEvents', None) != None \
-                        and getattr(fileName, 'events', None) != None and not getattr(fileName, 'merged', False):
-                        # Then direct-to-merge due to events if
-                        # the file is large enough:
-                        if fileName.events >= self.step.output.maxMergeEvents:
-                            # straight to merge
-                            try:
-                                fileName = self.handleLFNForMerge(mergefile = fileName, step = step)
-                            except Exception as ex:
-                                logging.error("Encountered error while handling LFN for merge due to events.\n")
-                                logging.error(str(ex))
-                                logging.debug(fileName)
-                                logging.debug("maxMergeEvents: %s" % self.step.output.maxMergeEvents)
-                                manager.cleanSuccessfulStageOuts()
-                                stepReport.addError(self.stepName, 60402, "DirectToMergeFailure", str(ex))
+
+                # Figuring out if we should do straight to merge
+                #  - should we do straight to merge at all ?
+                #  - is straight to merge disabled for this output ?
+                #  - are we over the size threshold
+                #  - are we over the event threshold ?
+                straightToMerge = False
+                if not getattr(fileName, 'merged', False) and hasattr(self.step.output, 'minMergeSize'):
+                    if fileName.module_label not in getattr(self.step.output, 'forceUnmergedOutputs', []):
+                        if hasattr(fileName, 'size'):
+                            if fileName.size >= self.step.output.minMergeSize:
+                                straightToMerge = True
+                        if hasattr(fileName, 'events') and hasattr(self.step.output, 'maxMergeEvents'):
+                            if fileName.events >= self.step.output.maxMergeEvents:
+                                straightToMerge = True
+
+                if straightToMerge:
+
+                    try:
+                        fileName = self.handleLFNForMerge(mergefile = fileName,
+                                                          step = step)
+                    except Exception as ex:
+                        logging.info("minMergeSize: %s", getattr(self.step.output, 'minMergeSize', None))
+                        logging.info("maxMergeEvents: %s", getattr(self.step.output, 'maxMergeEvents', None))
```

Comment:
I don't want to move these two lines up, since then I get that printout for every file that goes straight to merge. These two lines are were they are to help debugging of problems, ie. to give additional information when something fails.